shoes-core/lib/shoes/link.rb

```diff
@@ -18,9 +18,9 @@ def initialize(my_app, texts, styles = {}, blk = nil)
       super texts, @style
     end
 
-    # Force hovering evaluation up in parent where we have actual dimensions
-    def eval_in_parent?
-      true
+    # Hover must use containing text block since we're missing dimensions
+    def target
+      text_block
     end
```

Comment:
Tested this out, and the hover detection talks to the elements directly. The only thing the `target` is used against is the evaluation of the hover block. I just cribbed the dimensions comment from my earlier work, none of which explains when we need dimensions :(

That said, when I think about the odd nature of the `Link` and `Span` objects... they're not really Shoes elements, so having them be the `self` in the block evaluation might be surprising in a different sense.

I'm inclined to revise the comment but leave things as they are, or maybe check Shoes 3 to see what it has as the hover block evaluation target.